counsel.el

```diff
@@ -1322,34 +1322,38 @@ files in a project.")
        (setq cmd counsel-git-grep-cmd-default)))
     (cons proj cmd)))
 
-(defun counsel--call (&rest program-and-args)
-  (let ((stderr-file (make-temp-file "counsel-call-stderr")))
+(defun counsel--call (&rest command)
+  "Synchronously call COMMAND and return its output as a string.
+COMMAND comprises the program name followed by its arguments, as
+in `make-process'.  Signal `file-error' and emit a warning if
+COMMAND fails.  Obey file handlers based on `default-directory'."
+  (let ((stderr (make-temp-file "counsel-call-stderr-")))
     (unwind-protect
          (with-temp-buffer
-           (let ((res (apply #'call-process (car program-and-args)
-                             nil (list t stderr-file)
-                             nil (cdr program-and-args))))
-             (if (= 0 res)
-                 (buffer-substring (point-min)
-                                   (- (point-max)
-                                      (if (= ?\n (char-before (point-max)))
-                                          1
-                                        0)))
-               (error "%S exited with %d\n%s"
-                      (mapconcat #'identity program-and-args " ")
-                      res
-                      (with-temp-buffer
-                        (insert-file-contents stderr-file)
-                        (buffer-string))))))
-      (delete-file stderr-file))))
+           (let ((res (apply #'process-file (car command) nil
+                             (list t stderr) nil (cdr command))))
+             (unless (eq res 0)
+               (setq res (list 'file-error
+                               (mapconcat #'identity `(,@command "failed") " ")
+                               res))
+               (lwarn 'ivy :warning "%s"
+                      (apply #'concat
+                             (error-message-string res)
+                             (when (file-readable-p stderr)
+                               (insert-file-contents stderr nil nil nil t)
+                               (list "\n" (buffer-string)))))
+               (signal (car res) (cdr res)))
+             (buffer-substring (point-min) (- (point) (if (bolp) 1 0)))))
+      (when (file-exists-p stderr)
+        (delete-file stderr)))))
 
 (defun counsel--git-grep-count-func-default ()
   "Default function to calculate `counsel--git-grep-count'."
-  (if (eq system-type 'windows-nt) 0
-    (condition-case nil
-        (let ((git-dir (counsel--call "git" "rev-parse" "--git-dir")))
-          (read (counsel--call "du" "-s" git-dir)))
-      (error 0))))
+  (or (unless (eq system-type 'windows-nt)
```

Comment:
https://github.com/abo-abo/swiper/blob/986264097ff85a82cc9b369299a70167a47f2f62/counsel.el#L1214

It's size in bytes, not line count isn't it?  Perhaps change the documentation and rename it to `...-size` instead to convey the intent in more standard way?

Reply:
See my unanswered questions in #1827. I find the naming, documentation, and implementation of these features confusing, but I'm not sure how to proceed in changing them. Perhaps @abo-abo can advise.

Don't forget, though, that all of this counting is merely intended as a proxy, not as a hard and fast metric.